modules/apps/27-interchain-accounts/host/types/ack_test.go

```diff
@@ -43,41 +118,106 @@ func TestTypesTestSuite(t *testing.T) {
+	)
+
+	// Write consensus json to temp file, used for an IBC message.
+	// Generated by printing the result of cdc.MarshalIntefaceJSON on
+	// a solo machine consensus state
+	consensusJSON := testutil.WriteToNewTempFile(
+		suite.T(),
+		`{"@type":"/ibc.lightclients.solomachine.v2.ConsensusState","public_key":{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AtK50+5pJOoaa04qqAqrnyAqsYrwrR/INnA6UPIaYZlp"},"diversifier":"testing","timestamp":"10"}`,
+	)
+
+	testCases := []struct {
+		desc         string
+		cmd          *cobra.Command
+		args         []string
+		code         uint32
+		expDifferent bool
+	}{
+		{
+			"failing IBC message",
+			ibcclientcli.NewCreateClientCmd(),
+			[]string{
+				badClientStateJSON.Name(), // path to client state json
+				consensusJSON.Name(),      // path to consensus json,
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(suite.cfg.BondDenom, sdk.NewInt(10))).String()),
+				fmt.Sprintf("--gas=%d", flags.DefaultGasLimit),
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+				fmt.Sprintf("--%s=foobar", flags.FlagNote),
+			},
+			uint32(8),
+			true, // this transaction is expected to create a different ResultHash
 		},
 	}
 
-	deliverTxSameABCICode := sdkerrors.ResponseDeliverTx(errSameABCICode, gasUsed, gasWanted, false)
-	responsesSameABCICode := tmprotostate.ABCIResponses{
-		DeliverTxs: []*abcitypes.ResponseDeliverTx{
-			&deliverTxSameABCICode,
-		},
-	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, tc.cmd, tc.args)
+			suite.Require().NoError(err)
 
-	deliverTxDifferentABCICode := sdkerrors.ResponseDeliverTx(errDifferentABCICode, gasUsed, gasWanted, false)
-	responsesDifferentABCICode := tmprotostate.ABCIResponses{
-		DeliverTxs: []*abcitypes.ResponseDeliverTx{
-			&deliverTxDifferentABCICode,
-		},
+			var txRes sdk.TxResponse
+			suite.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txRes))
+			suite.Require().Equal(tc.code, txRes.Code)
+
+			deliverTxABCICode := abci.ResponseDeliverTx{
+				Codespace: txRes.Codespace,
+				Code:      txRes.Code,
+				Log:       txRes.RawLog,
+				GasWanted: txRes.GasWanted,
+				GasUsed:   txRes.GasUsed,
+			}
+
+			responses := tmprotostate.ABCIResponses{
+				DeliverTxs: []*abcitypes.ResponseDeliverTx{
+					&deliverTxABCICode,
+				},
+			}
+
+			expHash := ibctesting.ABCIResponsesResultsHash(&responses)
+
+			res, err := suite.network.Validators[0].RPCClient.BlockchainInfo(context.Background(), txRes.Height, txRes.Height)
+			suite.Require().NoError(err)
+
+			suite.Require().Equal(expHash, res.BlockMetas[0].Header.LastResultsHash.Bytes())
```

Comment:
I'm not sure why this is failing, I checked that `BlockMeta[0]` has the same height as `txRes`